torch/csrc/distributed/c10d/NCCLUtils.cpp

```diff
@@ -18,11 +18,20 @@ std::string getNcclVersion() {
     if (status != ncclSuccess || version < 100) {
       versionString = "Unknown NCCL version";
     } else {
-      auto ncclMajor = version / 1000;
-      auto ncclMinor = (version % 1000) / 100;
-      auto ncclPatch = version % (ncclMajor * 1000 + ncclMinor * 100);
+      // this condition is meant to mirror what is done in upstream NCCL
+      // e.g., 2.8.99 -> 2899, 2.9.0 -> 20900
+      int ncclMajor, ncclMinor, ncclPatch;
+      if (version >= 2900) {
+        ncclMajor = version / 10000;
+        ncclMinor = (version % 10000) / 100;
+        ncclPatch = version % (ncclMajor * 10000 + ncclMinor * 100);
+      } else {
```

Comment:
Can we have some tests for this?

Reply:
I'm not actually sure where tests for this would go, as the version numbers come from compile-time macros.

Reply:
CC @mrshenli

Reply:
e.g., doing something like `auto versionString = c10d::getNcclVersion();` in `test/cpp/c10d/ProcessGroupNCCLTest.cpp` just leads to a linking error on my end: 
```
ProcessGroupNCCLTest.cpp:(.text+0x26c): undefined reference to `c10d::getNcclVersion[abi:cxx11]()'
```